torchao/csrc/cuda/fp6_llm/fp6_linear.cu

```diff
@@ -153,6 +154,26 @@ cudaError_t fpx_linear_kernel(cudaStream_t    stream,
 #include <ATen/cuda/CUDAContext.h>
 #include <torch/library.h>
 
+// https://github.com/NVIDIA/apex/blob/master/csrc/type_shim.h
+// #if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 800
+#define DISPATCH_HALF_AND_BF16(TYPE, NAME, ...)                                \
+  switch (TYPE) {                                                              \
+  case at::ScalarType::Half: {                                                 \
+    using torch_t = at::Half;                                                  \
+    using nv_t = half;                                                         \
+    __VA_ARGS__();                                                             \
+    break;                                                                     \
+  }                                                                            \
+  case at::ScalarType::BFloat16: {                                             \
+    using torch_t = at::BFloat16;                                              \
+    using nv_t = __nv_bfloat16;                                                \
+    __VA_ARGS__();                                                             \
+    break;                                                                     \
+  }                                                                            \
```

Comment:
If we want to keep sm75 support, I think we need to be careful whenever we use `__nv_bfloat16`. Might need to add more `#ifdef`.